### PR TITLE
Updates to eavesdrop and eavesdrop_feature

### PR DESCRIPTION
**Pull request by extremerotary**
Hey guys,
The intention of these updates is to fix eavesdrop_feature and add an initial state of the eavesdrop mode on connect.

### CODE REVIEW

**Comment by extremerotary** [comment]
@jamesaimonetti 
I'm not quite sure why the circle CI checks are failing

**Comment by swysor** [comment]
> @jamesaimonetti
> I'm not quite sure why the circle CI checks are failing

You  need to make fmt on b/applications/callflow/src/module/cf_eavesdrop.erl (must install emacs for this to work) 

also appears make apis is failing, might want to try that locally to see what is goinng on there. Make sure you compile before running that.

File: applications/ecallmgr/src/call_cmd/ecallmgr_call_command.erl [hunk 1]
```diff
@@ -1514,3 +1515,47 @@ normalize_event_action_char(C) when is_integer(C), $A =< C, C =< $Z -> C + 32;
...
+            lager:debug("error bad value of Init-Whisper-Mode: ~p", [Other]),
+            'ok'
+    end.
+
+-spec eavesdrop_bridge_both(atom(), kz_term:ne_binary())  -> 'ok'.
+eavesdrop_bridge_both(Node, UUID) ->
+    _ = ecallmgr_util:send_cmd(Node, UUID, "unset", "eavesdrop_bridge_aleg"),
+    _ = ecallmgr_util:send_cmd(Node, UUID, "set", "eavesdrop_bridge_aleg=true"),
+    _ = ecallmgr_util:send_cmd(Node, UUID, "unset", "eavesdrop_bridge_bleg"),
+    _ = ecallmgr_util:send_cmd(Node, UUID, "set", "eavesdrop_bridge_bleg=true"),
```
Comment (lazedo) [inline applications/ecallmgr/src/call_cmd/ecallmgr_call_command.erl]: all these are expensive operations. it implies 4 calls to mod_kazoo, not a good thing

**Comment by extremerotary** [comment]
Okay will do, thanks

File: applications/ecallmgr/src/call_cmd/ecallmgr_call_command.erl [hunk 2]
```diff
@@ -1514,3 +1515,47 @@ normalize_event_action_char(C) when is_integer(C), $A =< C, C =< $Z -> C + 32;
...
 normalize_event_action_char(C) -> C.
+
+maybe_set_init_whisper_mode(Node, UUID, JObj) ->
+    case kz_json:get_ne_binary_value(<<"Init-Whisper-Mode">>, JObj) of
+        'undefined' -> 'ok';
+        Mode -> set_init_whisper_mode(Node, UUID, Mode)
+    end.
+
+-spec set_init_whisper_mode(atom(), kz_term:ne_binary(), binary())  -> 'ok'.
+set_init_whisper_mode(Node, UUID, Mode) ->
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/call_cmd/ecallmgr_call_command.erl]: Generally we would pattern match the Mode in the function clause:

```erlang
set_init_whisper_mode(Node, UUID, <<"a">>) -> ...;
set_init_whisper_mode(Node, UUID, <<"b">>) -> ...;
```

**Comment by neurosys-zero** [comment]
Hi! Great seeing some eavesdrop love. Do any of these commits address the case-sensitive issue with usernames/realms/devices names?

**Comment by neurosys-zero** [comment]
Surely! Easy enough to reproduce: go create a username for a device with a capital letter in it and it will break eavesdrop. Same for realms (not sure why monster started auto-generating caps in realms - but whatever) Also if a user enters a custom device username starting with that user's extension, it will break eavesdrop.

**Comment by extremerotary** [comment]
@lazedo Can you tell why the build failed?

**Comment by noahmehl** [comment]
@extremerotary It looks like some dialyzer checks are failing:

https://circleci.com/gh/2600hz/kazoo/33106?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link

Specifically:

```
/home/circleci/2600hz/kazoo/applications/omnipresence/src/kapi_omnipresence.erl:157: warn_umatched_return
  kapi_omnipresence.erl:157: Expression produces a value of type 'ok' | {'error',_}, but this value is unmatched

/home/circleci/2600hz/kazoo/applications/omnipresence/src/kapi_omnipresence.erl:166: warn_umatched_return
  kapi_omnipresence.erl:166: Expression produces a value of type 'ok' | {'error',_}, but this value is unmatched

/home/circleci/2600hz/kazoo/applications/omnipresence/src/kapi_omnipresence.erl:172: warn_umatched_return
  kapi_omnipresence.erl:172: Expression produces a value of type 'ok' | {'error',_}, but this value is unmatched
```

**Comment by jamesaimonetti** [comment]
@extremerotary you might want to check out @lazedo's comments on how master is separating things. As this PR is well behind master at this point, I'll close it; look forward to an updated PR if this functionality is still needed/useful.

**Pull request closed without merging**